extensions/ql-vscode/src/remote-queries/view/SortRepoFilter.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { FilterIcon } from '@primer/octicons-react';
+import { ActionList, ActionMenu } from '@primer/react';
+
+export type Sort = 'name' | 'stars' | 'results';
+type SortBy = { name: string, sort: Sort }[];
+type Props = {
+  sort: Sort;
+  setSort: (sort: Sort) => void;
+};
+
+type Sortable = {
+  nwo: string;
+  starCount?: number;
+  resultCount?: number;
+};
+
+const sortBy: SortBy = [
+  { name: 'Sort by Name', sort: 'name' },
+  { name: 'Sort by Results', sort: 'results' },
+  { name: 'Sort by Stars', sort: 'stars' },
+];
+
+export function sorter(sort: Sort) {
+  // stars and results are highest to lowest
+  // name is alphabetical
+  return (left: Sortable, right: Sortable) => {
+    if (sort === 'stars') {
+      const stars = (right.starCount || 0) - (left.starCount || 0);
+      if (stars !== 0) {
+        return stars;
+      }
+    }
+    if (sort === 'results') {
+      const results = (right.resultCount || 0) - (left.resultCount || 0);
+      if (results !== 0) {
+        return results;
+      }
+    }
+
+    // Fall back on name compare if results or stars are equal
+    return left.nwo.localeCompare(right.nwo, undefined, { sensitivity: 'base' });
+  };
+}
+
+const SortRepoFilter = ({ sort, setSort }: Props) => {
+  return <span className="vscode-codeql__analysis-sorter">
+    <ActionMenu>
```

Comment:
This doesn't look right locally for me for some reason 🤔 

<img width="1080" alt="image" src="https://user-images.githubusercontent.com/311693/169277128-8b08ead8-b7b8-4cab-965d-638ca95b4191.png">
<img width="1080" alt="image" src="https://user-images.githubusercontent.com/311693/169277178-44f7a6f7-810d-4477-88d6-c44cd8eb97c1.png">
<img width="1086" alt="image" src="https://user-images.githubusercontent.com/311693/169277349-1c73280d-15f9-43af-8505-0914f0cbf8ab.png">

Reply:
Actually this looked better after running `npm run build`. I'm not sure why, I thought `npm run watch` would refresh everything when it first runs.

There are some styling issues still though. 
- The colours are a bit off in light mode
- The menu opens up to the right - it should open to the left according to the designs
<img width="1233" alt="image" src="https://user-images.githubusercontent.com/311693/169278597-a1c9c2be-18d2-4cbb-b7e8-66eb07f63290.png">

Reply:
Don't know if i have control over where the drop-down comes. I'll try to fix.

I'll also work on the background color.

Reply:
The menu open location is dependent on the width of the viewport. With a smaller screen, the menu will open on the other side. It's the default behaviour of this component. Do you think this is something to spend time on?

Reply:
And another thing that's odd is that on my screen, I'm getting the reverse colours. I wonder if it has something to do with what theme the IDE is using when you initially run compile??

<img width="568" alt="_Extension_Development_Host__-_CodeQL_Query_Results_—_vscode-codeql-starter__Workspace_" src="https://user-images.githubusercontent.com/363559/169419464-24e77a5e-d986-4137-bc35-e1c32137eac2.png">

Odd. I'll have to look tomorrow.

Reply:
Discussed offline - we'll use the approach we have in the CodePaths component and work on our Primer theming solution.